### ISSUE TITLE
restart unless condition missmatches

### ISSUE THREAD

**Issue by brightdroid** [puppetlabs/puppetlabs-docker#469]
## What you expected to happen?

docker::run should start the container even if a image or network with the same name exists

## What happened?

container doesn't start when image or network with same name exists.

## How to reproduce it?

```puppet
  docker_network {'traefik':
  }
  docker::run {'traefik':
    image    => 'traefik',
    restart  => 'unless-stopped',
    net      => 'traefik',
  }
```

## Versions:
```
$ puppet --version
5.4.0

$ docker version
Client:
 Version:           18.09.4
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        d14af54266
 Built:             Wed Mar 27 18:35:44 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.4
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.8
  Git commit:       d14af54
  Built:            Wed Mar 27 18:01:48 2019
  OS/Arch:          linux/amd64
  Experimental:     false

$ facter os
{
  architecture => "amd64",
  distro => {
    codename => "bionic",
    description => "Ubuntu 18.04.2 LTS",
    id => "Ubuntu",
    release => {
      full => "18.04",
      major => "18.04"
    }
  },
  family => "Debian",
  hardware => "x86_64",
  name => "Ubuntu",
  release => {
    full => "18.04",
    major => "18.04"
  },
  selinux => {
    enabled => false
  }
}

$ puppet module list
```

**Comment by brightdroid** [comment]
### Patch
```patch
diff --git a/manifests/run.pp b/manifests/run.pp
index db87c09..f087e36 100755
--- a/manifests/run.pp
+++ b/manifests/run.pp
@@ -239,7 +239,7 @@ define docker::run(
     $exec_provider = 'powershell'
     $cidfile = "${::docker_user_temp_path}/${service_prefix}${sanitised_title}.cid"
 # lint:ignore:140chars
-    $restart_check = "${docker_command} inspect ${sanitised_title} -f '{{ if eq \\\"unhealthy\\\" .State.Health.Status }} {{ .Name }}{{ end }}' | findstr ${sanitised_title}"
+    $restart_check = "${docker_command} container inspect ${sanitised_title} -f '{{ if eq \\\"unhealthy\\\" .State.Health.Status }} {{ .Name }}{{ end }}' | findstr ${sanitised_title}"
 # lint:endignore
   } else {
     $exec_environment = 'HOME=/root'
@@ -248,7 +248,7 @@ define docker::run(
     $exec_provider = undef
     $cidfile = "/var/run/${service_prefix}${sanitised_title}.cid"
 # lint:ignore:140chars
-    $restart_check = "${docker_command} inspect ${sanitised_title} -f '{{ if eq \"unhealthy\" .State.Health.Status }} {{ .Name }}{{ end }}' | grep ${sanitised_title}"
+    $restart_check = "${docker_command} container inspect ${sanitised_title} -f '{{ if eq \"unhealthy\" .State.Health.Status }} {{ .Name }}{{ end }}' | grep ${sanitised_title}"
 # lint:endignore
   }
 
@@ -267,7 +267,7 @@ define docker::run(
     if $ensure == 'absent' {
       exec { "stop ${title} with docker":
         command     => "${docker_command} stop --time=${stop_wait_time} ${sanitised_title}",
-        onlyif      => "${docker_command} inspect ${sanitised_title}",
+        onlyif      => "${docker_command} container inspect ${sanitised_title}",
         environment => $exec_environment,
         path        => $exec_path,
         provider    => $exec_provider,
@@ -276,7 +276,7 @@ define docker::run(
 
       exec { "remove ${title} with docker":
         command     => "${docker_command} rm -v ${sanitised_title}",
-        onlyif      => "${docker_command} inspect ${sanitised_title}",
+        onlyif      => "${docker_command} container inspect ${sanitised_title}",
         environment => $exec_environment,
         path        => $exec_path,
         provider    => $exec_provider,
@@ -293,7 +293,14 @@ define docker::run(
         "--name ${sanitised_title} --cidfile=${cidfile}",
         "--restart=\"${restart}\" ${image} ${command}",
       ]
-      $inspect = ["${docker_command} inspect ${sanitised_title}"]
+
+      if $sanitised_title == 'mayan_app' {
+        notify{$sanitised_title:
+          message => $run_with_docker_command,
+        }
+      }
+
+      $inspect = ["${docker_command} container inspect ${sanitised_title}"]
       if $custom_unless {
         $exec_unless = concat($custom_unless, $inspect)
       } else {
@@ -311,7 +318,7 @@ define docker::run(
       if $running == false {
         exec { "stop ${title} with docker":
           command     => "${docker_command} stop --time=${stop_wait_time} ${sanitised_title}",
-          unless      => "${docker_command} inspect ${sanitised_title} -f \"{{ .State.Running }}\" | grep true",
+          unless      => "${docker_command} container inspect ${sanitised_title} -f \"{{ .State.Running }}\" | grep true",
           environment => $exec_environment,
           path        => $exec_path,
           provider    => $exec_provider,
@@ -320,7 +327,7 @@ define docker::run(
       } else {
         exec { "start ${title} with docker":
           command     => "${docker_command} start ${sanitised_title}",
-          onlyif      => "${docker_command} inspect ${sanitised_title} -f \"{{ .State.Running }}\" | grep false",
+          onlyif      => "${docker_command} container inspect ${sanitised_title} -f \"{{ .State.Running }}\" | grep false",
           environment => $exec_environment,
           path        => $exec_path,
           provider    => $exec_provider,
@@ -371,7 +378,7 @@ define docker::run(
         exec {
           "stop container ${service_prefix}${sanitised_title}":
           command     => "${docker_command} stop --time=${stop_wait_time} ${sanitised_title}",
-          onlyif      => "${docker_command} inspect ${sanitised_title}",
+          onlyif      => "${docker_command} container inspect ${sanitised_title}",
           environment => $exec_environment,
           path        => $exec_path,
           provider    => $exec_provider,
@@ -389,7 +396,7 @@ define docker::run(
       exec {
         "remove container ${service_prefix}${sanitised_title}":
         command     => "${docker_command} rm -v ${sanitised_title}",
-        onlyif      => "${docker_command} inspect ${sanitised_title}",
+        onlyif      => "${docker_command} container inspect ${sanitised_title}",
         environment => $exec_environment,
         path        => $exec_path,
         refreshonly => true,

```

**Comment by brightdroid** [comment]
Can you please update the commands as this is not precise enough when you have e.g. an image and container with the same name. Thanks

**Comment by adrianiurca** [comment]
work releated for this issue is going to be included in PR #705
